environments/dataset_generator.py

```diff
@@ -82,22 +81,32 @@ def env_thread(args, thread_num, partition=True, use_ppo2=False):
                (thread_num if thread_num <= args.num_episode % args.num_cpu else args.num_episode % args.num_cpu)
 
         env.seed(seed)
-        prng.seed(seed)  # this is for the sample() function from gym.space
+        env.action_space.seed(seed)  # this is for the sample() function from gym.space
         obs = env.reset()
         done = False
         t = 0
+        episode_toward_target_on = False
         while not done:
             env.render()
 
             if use_ppo2:
                 action, _ = model.predict([obs])
             else:
-                action = [env.action_space.sample()]
+                if episode_toward_target_on and np.random.rand() < args.toward_target_timesteps_proportion:
+                    action = [env.actionPolicyTowardTarget()]
```

Comment:
Is that implemented for all envs ?

Reply:
File updated! it's specific the case of data-collection on the Real Omnibot